flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/TableSinkFactory.java

```diff
@@ -53,4 +55,39 @@
 		return createTableSink(table.toProperties());
 	}
 
+	/**
+	 * Creates and configures a {@link TableSink} based on the given
+	 {@link Context}.
+	 *
+	 * @param context context of this table sink.
+	 * @return the configured table sink.
+	 */
+	default TableSink<T> createTableSink(Context context) {
```

Comment:
Should we deprecate the other `createTableSink` and `createTableSource` interfaces?

Reply:
I don't know... we can deprecate `createTableSink(ObjectPath, CatalogTable)`. CC: @twalthr